client/src/modules/list/components/Items/ItemsList.jsx

```diff
@@ -5,33 +5,61 @@ import { FormattedMessage } from 'react-intl';
 import { withRouter } from 'react-router-dom';
 import { CSSTransition, TransitionGroup } from 'react-transition-group';
 import _flowRight from 'lodash/flowRight';
+import _isEqual from 'lodash/isEqual';
 
 import ListItem from 'modules/list/components/Items/ListItem';
 import ListArchivedItem from 'modules/list/components/Items/ListArchivedItem';
 import MessageBox from 'common/components/MessageBox';
 import { MessageType } from 'common/constants/enums';
 import { disableItemAnimations } from './model/actions';
 import { RouterMatchPropType } from 'common/constants/propTypes';
-import { updateLimit } from 'modules/list/model/actions';
 import { DISPLAY_LIMIT } from 'common/constants/variables';
 
 class ItemsList extends PureComponent {
   constructor(props) {
     super(props);
 
     this.state = {
+      displayedItems: null,
       limit: DISPLAY_LIMIT
     };
   }
 
+  componentDidMount() {
+    this.handleItems();
+  }
+
+  componentDidUpdate(previousProps) {
+    const { items: previousItems } = previousProps;
+    const { items } = this.props;
+
+    if (!_isEqual(previousItems, items)) {
```

Comment:
Items were an array before. Is it necessary for them to be an object? Can we use array and compare items length? Isn't it simpler and faster? And without external libs?